ci/mkpipeline.py

```diff
@@ -240,6 +243,32 @@ def visit(step: dict[str, Any]) -> None:
                 visit(inner_step)
 
 
+def trim_test_selection(pipeline: Any, steps_to_run: set[str]) -> None:
+    def visit(step: dict[str, Any]) -> None:
+        if (
+            step.get("id") not in steps_to_run
+            and "prompt" not in step
+            and "wait" not in step
+            and "group" not in step
```

Comment:
Do you also want to match entire groups?

Reply:
No, instead each test in the group is separately selectable.